gallery/gallery-war/src/main/java/org/silverpeas/components/gallery/web/AbstractGalleryResource.java

```diff
@@ -186,24 +204,108 @@ protected Response getMediaContent(final MediaType expectedMediaType, final Stri
+      return Pair.of(media, file);
+    } catch (final WebApplicationException ex) {
+      throw ex;
+    } catch (final Exception ex) {
+      throw new WebApplicationException(ex, Status.SERVICE_UNAVAILABLE);
+    }
+  }
+
+  /**
+   * Centralization of getting of media content.
+   * @param expectedMediaType expected media type
+   * @param mediaId the media identifier
+   * @param requestedMediaResolution requested media resolution
+   * @return the response.
+   */
+  Response getMediaContent(final MediaType expectedMediaType, final String mediaId,
+      final MediaResolution requestedMediaResolution) {
+    try {
+      final Pair<Media, SilverpeasFile> checkedMedia =
+          getCheckedMedia(expectedMediaType, mediaId, requestedMediaResolution);
+      final Media media = checkedMedia.getLeft();
+      final SilverpeasFile file = checkedMedia.getRight();
+
+      java.nio.file.Path mediaPath = Paths.get(file.toURI());
+      String mediaMimeType = getHttpRequest().getParameter("forceMimeType");
+      if (StringUtil.isNotDefined(mediaMimeType)) {
+        mediaMimeType = ((InternalMedia) media).getFileMimeType().getMimeType();
+      }
+
+      int length = (int) Files.size(mediaPath);
+      int start = 0;
+      int end = length - 1;
+
+      String range = defaultStringIfNotDefined(getHttpServletRequest().getHeader("Range"), "");
+      Matcher matcher = RANGE_PATTERN.matcher(range);
+      boolean isPartialRequest = matcher.matches();
+
+      if (isPartialRequest) {
+        String startGroup = matcher.group("start");
+        start = startGroup.isEmpty() ? start : Integer.valueOf(startGroup);
+        start = start < 0 ? 0 : start;
+
+        String endGroup = matcher.group("end");
+        end = endGroup.isEmpty() ? end : Integer.valueOf(endGroup);
+        end = end > length - 1 ? length - 1 : end;
+      }
+
+      final int finalStart = start;
+      final int finalContentLength = end - finalStart + 1;
+
+      final Response.ResponseBuilder responseBuilder;
+      if (isPartialRequest) {
+
+        // Handling here a partial response (pseudo streaming)
+
+        final HttpServletResponse response = getHttpServletResponse();
+        response.setBufferSize(BUFFER_LENGTH);
+
+        responseBuilder =
+            Response.status(Status.PARTIAL_CONTENT).entity((StreamingOutput) output -> {
+
+              int bytesRead;
+              int bytesLeft = finalContentLength;
+              ByteBuffer buffer = ByteBuffer.allocate(BUFFER_LENGTH);
+
+              try (SeekableByteChannel input = Files.newByteChannel(mediaPath, READ)) {
+                input.position(finalStart);
+                while ((bytesRead = input.read(buffer)) != -1 && bytesLeft > 0) {
+                  buffer.clear();
+                  output.write(buffer.array(), 0, bytesLeft < bytesRead ? bytesLeft : bytesRead);
+                  bytesLeft -= bytesRead;
+                }
+              } catch (IOException ioe) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)